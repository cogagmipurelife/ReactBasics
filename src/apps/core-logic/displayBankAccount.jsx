
import React, {Component} from 'react';
import TestAccount from '../axios/testAccount';
import DataTable from '../core-logic/DataTable';
import displayBankAccountcss from '../core-logic/displayBankAccount.css'

export default class displayBankAccount extends React.Component
{

    
    constructor(props) {
        super(props);
        this.state = { AccountsData: [] };
    }

   async componentDidMount()
    {
        await TestAccount.get().then(res => {
            this.setState({ AccountsData: res.data.bankList });
        })  
        .catch(function (error) {
            console.log(error);
        })
    
        
}


dataTable() {
    return this.state.AccountsData.map((data, i) => {
        return <DataTable obj={data} key={i} />;
    });
}
render() {
    return (
        <div className="wrapper-users">
            <div className="container">
                <table className="table table-striped">
                    <thead className="table-dark">
                        <tr>
                            <td>Account Type</td>
                            <td>Bank Aocount Number</td>
                            <td>Bank Routing Number</td>
                            <td>Bank Name </td>
                        </tr>
                    </thead>
                    <tbody className='table-light displayBankAccountcss'>
                        {this.dataTable()}
                    </tbody>
                </table>
            </div>
        </div>
    )
}


}