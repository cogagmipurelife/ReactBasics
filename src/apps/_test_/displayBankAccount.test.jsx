import React from 'react'
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import displayBankAccount from '../core-logic/displayBankAccount';
Enzyme.configure({adapter: new Adapter()});

describe ('displayBankAccount' , () =>
{
    it('should render correctly', () =>
    {
        const component = shallow(<displayBankAccount/>);
        expect(component).toMatchSnapshot();
    });
    
});
