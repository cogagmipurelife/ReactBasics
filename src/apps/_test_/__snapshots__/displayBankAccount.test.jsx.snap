// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displayBankAccount should render correctly 1`] = `ShallowWrapper {}`;
